### PR TITLE
Fix #156, various parameter issues

### LINKED ISSUE #156: Various issues with converting v2 -> v3 with parameters
Hi,

I've found a few issues with converting parameters across from v2 to v3:
* $refs don't convert at all
* path and global references don't convert across

(PR incoming)

### PR DESCRIPTION
**Pull request by anz-gordonj7**
* Params with references are converted correctly
* Global and Path params are converted across

### CODE REVIEW

File: openapi2conv/openapi2_conv_test.go [hunk 1]
```diff
@@ -1,11 +1,10 @@
-package openapi2conv_test
+package openapi2conv
```
Comment (fenollp) [inline openapi2conv/openapi2_conv_test.go]: Why the test package rename?

File: openapi2conv/openapi2_conv.go [hunk 2]
```diff
@@ -364,6 +370,7 @@ func FromV3Swagger(swagger *openapi3.Swagger) (*openapi2.Swagger, error) {
 		if pathItem == nil {
 			continue
 		}
+		result.AddOperation(path, "GET", nil)
```
Comment (fenollp) [inline openapi2conv/openapi2_conv.go]: What does this solve?

File: openapi2conv/openapi2_conv_test.go (hunk 1 above)
Comment (anz-gordonj7) [reply openapi2conv/openapi2_conv_test.go]: avoids needing to import the real package, no need to put the tests in a different package name

File: openapi2conv/openapi2_conv.go (hunk 2 above)
Comment (anz-gordonj7) [reply openapi2conv/openapi2_conv.go]: Some of the swagger files I was converting have paths defined with params but no operations, im not sure if that is even sensible swagger, but adding this call means that the call to `result.Paths[path].Parameters = params` is guaranteed safe, without having to check that the maps are initialised. The nil "GET" op becomes a no-op anyway

Comment (fenollp) [reply openapi2conv/openapi2_conv.go]: This project is meant to support only the spec. Please fix your swagger files so they are valid and revert this.

File: openapi2conv/openapi2_conv_test.go (hunk 1 above)
Comment (fenollp) [reply openapi2conv/openapi2_conv_test.go]: This makes sure no test code lingers in the actual package. Please revert.

**Pull request merged**